### ISSUE TITLE
Make user input more "script" friendly

### ISSUE THREAD

**Issue by raff** [GaryBrownEEngr/turtle#1]
First of all, great work! I was playing with the idea of a minimal lisp interpreter to use for scripting, and I needed an example to test out how to extend it. I found your "turtle" library and came out with this: https://github.com/raff/gisp/tree/main/cmd/turtle

Everything work great, but I found that the interface of Canvas.GetUserInput is not the most friendly for my use case. See the implementation of the "pressed" method (https://github.com/raff/gisp/blob/main/cmd/turtle/main.go#L39) and "justPressed" (https://github.com/raff/gisp/blob/main/cmd/turtle/main.go#L115).

Basically I am trying to do something similar to what ebitengine does (in ebiten.IsKeyPressed or inpututil.IsKeyJustPressed) where you pass the key name (or character) but I need to convert from that to the UserInput format, than then converts back to ebiten format.

Would you be open to support something closer to what ebiten provides ? Maybe as an alternative ?

Also, it would be good if it was possible to terminate the application (by returning ebiten.Termination from the Update method). See for example https://github.com/raff/ebi-games/blob/main/minesweeper/main.go#L348.

I was going to try to implement it, by adding a Terminate() method and a terminate boolean to the drawCmd struct, but I got stuck thinking on which model to attach it to.

Another option, that maybe would solve both my problems, would be to have the option to provide a callback that gets executed in the game Update method if available. That way I could implement my own input processing and return the appropriate code if I want to terminate the application.

**Comment by GaryBrownEEngr** [comment]
Thanks for taking the time to reach out. I was able to easily back-port some of the improvements I made to the user input system for my golang scratch project. It is unfortunately a breaking change, but I am still in version 0, so it is to be expected. A new tag has been created: `v0.5.0`. It has the following improvements:
- new `Exit()` method for the canvas interface.
- A way to look up if a key is pressed using the `func (s *UserInput) IsPressedByName(name string) bool` method
- A way to subscribe to a data broker that provides just pressed events through a channel.
These all can be seen in use in the example `drawwithmouse`. https://github.com/GaryBrownEEngr/turtle/blob/main/examples/drawwithmouse/main.go

The just pressed key information has to be channel based because there is no guarantee that the turtle is running fast enough to always pick up all the just pressed events. But Ebiten requires that you check for the just pressed on every update cycle otherwise you will miss it.
